tools/test.sh

```diff
@@ -0,0 +1,55 @@
+#! /usr/bin/env bash
+
+# Copyright 2017 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+#     Unless required by applicable law or agreed to in writing, software
+#     distributed under the License is distributed on an "AS IS" BASIS,
+#     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#     See the License for the specific language governing permissions and
+#     limitations under the License.
+
+# This script will run the integration tests for the given app.
+#   $1, the path to the root of the app.
+#   $2, the project to use.
+
+# Exit on error or undefined variable
+set -eu
+
+if [ -z "${1:-}" ]; then
+    echo "Must specify the root of the app."
+    exit 1
+fi
+
+# If no project is given get the ambient project.
+if [ -z "${2:-}" ]; then
+    readonly project_id=$(gcloud config list core/project --format="csv[no-heading](core)" | cut -f 2 -d '=')
+    echo "Warning: Using ambient project ${project_id}"
+else
+    readonly project_id=$2
+fi
+
+# Where to store the modified app definitions.
+readonly temp_builders_root=$(mktemp -d -t test_run)
+
+# Generate the builders root.
+cp $1/runtimes.yaml ${temp_builders_root}
+export readonly STAGING_BUILDER_IMAGE=gcr.io/aspnetcore-staging/aspnetcorebuild:latest
```

Comment:
Usually I would say yes, but you have a good point, I added `TAG` as away to override the tag for the builder.